src/task1/plot_pois/plot_pois.py

```diff
@@ -52,52 +58,70 @@
 	pin_image_height = pin_image.shape[0];
 	pin_image_alpha = pin_image[:,:,3] /255.0;	
 	im_image_alpha = 1.0 - pin_image_alpha;
-	print(pin_image_alpha.shape);
-	print(im_image_alpha.shape);
-	cv.imshow('',pin_image);
-	cv.waitKey(0);		
 except Exception as e:
-	print('Could not lead and manipulate pin image.');
+	print('Could not load and manipulate pin image.');
 	print(e);
 	exit();
 
-print('Plot POIs Configuration Setup:');
+print('Plot POIs (v2) Configuration Setup:');
 print('-------------------------');
 print('Map file:', im_file);
 print('Damaged file:', id_file);
 print('Output file:',output_file);
-97
+
 # Read the file into 
 im_img = cv.imread(im_json['filename']);
 im_imgsize = im_img.shape;
 im_width = im_imgsize[1];
 im_height = im_imgsize[0];
 
-im_tl = im_json['topleft'];
-im_br = im_json['bottomright'];
-dx = im_br['long'] - im_tl['long'];
-dy = im_tl['lat'] - im_br['lat'];
+# Construct a fit using the given points
+# The region we are dealing with is small, each dimension is 6.65 millionth of the total radius @ 45deg latitude
+# So assume that a linear fit is accurate, but our points have some error
+# Thus, we do a linear regression
+given_lats = [];
+given_x = [];
+given_longs = [];
+given_y = [];
+for point in im_json['points']:
+	given_lats.append(point['lat']);
+	given_x.append(point['x']);
+	given_longs.append(point['long']);
+	given_y.append(point['y']);
+slope_x, intercept_x, r_value_x, p_value_x, std_err_x = stats.linregress(given_longs,given_x);
+slope_y, intercept_y, r_value_y, p_value_y, std_err_y = stats.linregress(given_lats,given_y);
+long2x = poly([intercept_x, slope_x])
+lat2y = poly([intercept_y,slope_y]);
 
-# For each coordinate position we have
+# Now take the GPs coordinates of the damaged panels and
 for coords in id_json['damaged']:
-	# place a marker 
 	lat = coords['lat'];
 	long = coords['long'];
+	msg = coords['message'];
 	
-	# pixel along x
-	px = ((long - im_tl['long'])/dx) * im_width;
-	py = ((im_tl['lat'] - lat)/dy) * im_height;
+	# Convert them to their pixel values
+	px = long2x(long);
+	py = lat2y(lat);
 	
 	# at x,y the image is drawn to the right and then bottom
 	# So we go -width/2 and -height
 	pxd = int(px - pin_image_width/2);
 	pxd2 = int(pxd + pin_image_width);
 	pyd = int(py - pin_image_height);
-	pyd2 = int(pyd + pin_image_height);	
+	pyd2 = int(pyd + pin_image_height);		
+	
+	
 	for chan in range(0,3): 
```

Comment:
I would see if theres a built in function that can do this all at once... otherwise this is fine